### PR TITLE
Fix bug where flows with names that do not match the function name cannot not be loaded

### PR DESCRIPTION
**Pull request by peytonrunyan**
Closes https://github.com/PrefectHQ/prefect/issues/7918

The `select_flows` function was checking to see if flow names found in a module matched a desired function name. This was causing problems in two cases:
1. the flow function had underscores, which were converted to dashes in the flow name
2. the flow had a custom name, which would not relate to the underlying function

This is now addressed by checking the underlying function name. 

The parameter `flow_name` in `select_flows` is now also sanitized to make sure that any dashes are converted to underscores, so that it will match the underlying flow function name.  

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by anna-geller** [comment]
works - QA
image

**Review by anna-geller — APPROVED** [review]
thanks so much for fixing it so quickly!

image

File: src/prefect/flows.py [hunk 1]
```diff
@@ -680,12 +680,13 @@ def select_flow(
         MissingFlowError: If a flow name is provided and that flow does not exist
         UnspecifiedFlowError: If multiple flows exist but no flow name was provided
     """
-    # Convert to flows by name
-    flows = {f.name: f for f in flows}
-
+    # We search for the flow's underlying function, so any dashes must be
+    #   replaced with underscores
+    flow_name = flow_name.replace("-", "_")
```
Comment (madkinsz) [inline src/prefect/flows.py:685]: Let's drop these lines.

Comment (madkinsz) [reply src/prefect/flows.py:685]: Oh maybe not? I'm a little confused.

**Comment by madkinsz** [comment]
@anna-geller we've changed the implementation of the fix if you want to QA again

File: src/prefect/cli/deployment.py [hunk 2]
```diff
@@ -752,6 +752,8 @@ async def build(
     try:
         flow = await run_sync_in_worker_thread(load_flow_from_entrypoint, entrypoint)
         app.console.print(f"Found flow {flow.name!r}", style="green")
+    except AttributeError as exc:
+        exit_with_error(f"{obj_name!r} not found in {fpath!r}.")
```
Comment (madkinsz) [inline src/prefect/cli/deployment.py:756]: Isn't `load_flow_from_entrypoint` guaranteed not to raise an `AttributeError`?

File: tests/cli/deployment/test_deployment_build.py [hunk 3]
```diff
@@ -535,9 +536,53 @@ def fn():
         res = invoke_and_assert(
             cmd,
             expected_code=1,
-            expected_output_contains=f"The provided path does not point to a python file: {str(fpath)!r}",
+            # expected_output_contains=f"The provided path does not point to a python file: {str(fpath)!r}",
```
Comment (madkinsz) [inline tests/cli/deployment/test_deployment_build.py:539]: @peytonrunyan

**Pull request merged**